tests/change-key-removes-user.sh

```diff
@@ -3,7 +3,11 @@ set -euo pipefail
 
 tmp=/root/tmp
 mkdir -p $tmp
-trap 'rm -rf $tmp; kill -- -$$' EXIT
+cleanup() {
+    kill -9 $server_pid || true
+    rm -rf $tmp
+}
+trap cleanup EXIT
```

Comment:
What's the motivation for this change? Is it a bugfix?

Reply:
Sorry, there is a comment on the PR, but python3.5's http.server doesn't have --directory. And as the tests run inside the docker container, the tests *also* need to support py35, sadly.

Reply:
Sorry, I don't understand. I saw the change to directory handling below, but this doesn't seem related. Am I missing something?

Reply:
gah early reply.

This meant changing how the python -m http.server was run in the background, from: `python -m ... &` to `(cd $tmp && python -m ...)&

For some reason, `kill -- -$$` (kill all subprocesses) was erroring with 143 (128+SIGTERM, so it errored while handling SIGTERM) when I switch to using the bash (subshell)& syntax. Still not sure why. Killing the specific PID worked (and is cleaner anyway).

But expanding the clean up to a function was part of the debugging process, and I felt it was cleaner, so left it in. Killing the process *then* removing the dir is probably a cleaner order to do it too.

Normally, I'd use` env --chdir python -m ... &`, but debian stretch has a very old version of coreutils, which doesn't support `env --chdir`.

Reply:
Thank you. I agree that the new version is better in all sorts of ways.